src/main/java/com/technicjelle/Cheese.java

```diff
@@ -173,20 +174,24 @@ public static Cheese createFromCellsUnsafe(Vector2d cellSize, Vector2i... cells)
 	 * @see #createMultiCheeseFromChunks(Vector2i...)
 	 */
 	public static boolean checkConnected(Vector2i... cells) {
-		Set<Vector2i> cellsToCheck = new HashSet<>(List.of(cells));
+		if (cells.length == 0) return false;
+		if (cells.length == 1) return true;
+
+		Set<Vector2i> cellsToCheck = Set.of(cells);
 		Set<Vector2i> visited = new HashSet<>();
-		Queue<Vector2i> toVisit = new LinkedList<>();
-		toVisit.add(cellsToCheck.iterator().next()); // start with the first cell
+		Stack<Vector2i> toVisit = new Stack<>();
```

Comment:
Tbh, the performance gains might just be stack vs linked list, what about stack vs an actual queue? 
Just a consideration. Still LGTM :)